### DISCUSSION TITLE
Please add window geometry information to 'niri msg focused-window'

### DISCUSSION THREAD

**Discussion by kevinclevenger** [YaLTeR/niri#721] [Ideas]
It would save some time fiddling around getting the window size exactly correct for 'default-column-width { fixed ... ; }'

**Comment by YaLTeR** [comment]
Yeah, window size (as in geometry size) should be added to the IPC window struct

**Reply by LawnGnome** [reply]
I started playing around with this, since I could also use the geometry for a project I'm working on. Getting the toplevel size is easy enough in `ipc::server::make_ipc_window`, since it's available on the `ToplevelState`, but I'm struggling to figure out how to get the toplevel location relative to the workspace.

Do you have any pointers on what I could look at there?

Thanks!

**Reply by YaLTeR** [reply]
@calops is working on it

**Reply by nivaddo** [reply]
any updates on this? as im quite interested in getting the geometry size in `niri msg windows` as well for a project

**Reply by YaLTeR** [reply]
Latest work is here: https://github.com/YaLTeR/niri/pull/1265

**Reply by nivaddo** [reply]
Thanks, looking forward to it. Amazing work!
